openai/openai.go

```diff
@@ -148,7 +149,33 @@ func toChunk(id string, r api.ChatResponse) ChatCompletionChunk {
 func fromRequest(r ChatCompletionRequest) api.ChatRequest {
 	var messages []api.Message
 	for _, msg := range r.Messages {
-		messages = append(messages, api.Message{Role: msg.Role, Content: msg.Content})
+		switch content := msg.Content.(type) {
+		case string:
+			messages = append(messages, api.Message{Role: msg.Role, Content: content})
+		case []any:
+			for _, c := range content {
+				if data, ok := c.(map[string]any); ok {
+					message := api.Message{Role: msg.Role}
+					switch data["type"] {
+					case "text":
+						if text, ok := data["text"].(string); ok {
+							message.Content = text
+						}
+					case "image_url":
+						if urlMap, ok := data["image_url"].(map[string]any); ok {
+							if url, ok := urlMap["url"].(string); ok {
+								if img, err := base64.StdEncoding.DecodeString(url); err == nil {
```

Comment:
I think we need to remove the `data:image/jpeg;base64,` prefix here:

```
strings.TrimPrefix(url, "data:image/jpeg;base64,")
```

Source: https://platform.openai.com/docs/guides/vision/uploading-base-64-encoded-images

Reply:
Great catch, thanks

Reply:
🙌